### PR TITLE
Remove inbox validation for uploading key packages

### PR DESCRIPTION
**Pull request by neekolas**
## tl;dr
- Implements part of Phase 1 of https://github.com/xmtp/xmtp-node-go/issues/399
- Goal here is to loosen the validations on uploading key packages so that uploads can happen before `PublishIdentityUpdate` is called
- Begins the process of cleaning up these APIs and simplifying things

## AI Assisted Summary
This pull request includes several changes to the project, primarily focused on refactoring the installation registration process and updating the database schema. The key changes made are:
1. Removed `inbox_id` and `expiration` columns from the `installations` table.
2. Updated SQL queries to reflect the schema changes, including the removal of `UpdateKeyPackage` query and the addition of `CreateOrUpdateInstallation` query with upsert logic.
3. Deprecated the `RegisterInstallation` method in favor of `UploadKeyPackage`.
4. Updated test cases to align with the new schema and methods.
5. Simplified the `ValidateInboxIdKeyPackages` method by removing intermediate validation requests and leveraging direct validation responses.

These changes aim to streamline the installation registration process, reduce database schema complexity, and improve overall code maintainability.

---

### CODE REVIEW

**Comment by neekolas** [comment]
* **#400** <a href="https://app.graphite.dev/github/pr/xmtp/xmtp-node-go/400?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @neekolas and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: dev/docker/env [hunk 1]
```diff
@@ -2,5 +2,5 @@
 set -e
 
 function docker_compose() {
-  docker-compose -f dev/docker/docker-compose.yml -p xmtpd "$@"
+  docker compose -f dev/docker/docker-compose.yml -p xmtpd "$@"
```
Comment (neekolas) [inline dev/docker/env:5]: Apparently our Github runners have been upgraded to a new version where `docker-compose` no longer exists as a separate command. You must use `docker compose` instead.

File: pkg/api/message/v1/service.go [hunk 2]
```diff
@@ -130,17 +130,17 @@ func (s *Service) Publish(ctx context.Context, req *proto.PublishRequest) (*prot
 		log.Debug("received message")
 
 		if len(env.ContentTopic) > MaxContentTopicNameSize {
-			return nil, status.Errorf(codes.InvalidArgument, "topic length too big")
+			return nil, status.Error(codes.InvalidArgument, "topic length too big")
```
Comment (neekolas) [inline pkg/api/message/v1/service.go:133]: Our linter has apparently become more fussy with some version upgrade and this old code was causing lint fails

File: pkg/api/interceptor.go [hunk 3]
```diff
@@ -96,17 +96,17 @@ func (wa *WalletAuthorizer) requiresAuthorization(req interface{}) bool {
 func (wa *WalletAuthorizer) getWallet(ctx context.Context) (types.WalletAddr, error) {
 	md, ok := metadata.FromIncomingContext(ctx)
 	if !ok {
-		return "", status.Errorf(codes.Unauthenticated, "metadata is not provided")
+		return "", status.Error(codes.Unauthenticated, "metadata is not provided")
```
Comment (neekolas) [inline pkg/api/interceptor.go:99]: Something has been updated in our linter and it seems to have become stricter on using `Errorf` for errors that don't require formatting

File: pkg/mls/api/v1/service.go [hunk 4]
```diff
@@ -113,6 +113,11 @@ func (s *Service) HandleIncomingWakuRelayMessage(wakuMsg *wakupb.WakuMessage) er
 	return nil
 }
 
+/*
+*
+DEPRECATED: Use UploadKeyPackage instead
+*
+*/
 func (s *Service) RegisterInstallation(ctx context.Context, req *mlsv1.RegisterInstallationRequest) (*mlsv1.RegisterInstallationResponse, error) {
```
Comment (neekolas) [inline pkg/mls/api/v1/service.go:121]: We don't really need a separate endpoint for registering an installation and updating the key package. Can just do a "create or update" query and have one endpoint for everything.

**Review by richardhuaaa — APPROVED** [review]
Love it, this is so much cleaner

File: pkg/mlsvalidate/service.go [hunk 5]
```diff
@@ -87,11 +87,6 @@ func (s *MLSValidationServiceImpl) GetAssociationState(ctx context.Context, oldU
 	}, nil
 }
 
-type KeyAndExpiration struct {
-	InstallationId []byte
-	Expiration     uint64
-}
-
 func (s *MLSValidationServiceImpl) ValidateInboxIdKeyPackages(ctx context.Context, keyPackages [][]byte) ([]InboxIdValidationResult, error) {
```
[the remark this answers is not in the document]
Comment (neekolas) [reply pkg/mlsvalidate/service.go:90]: I have some items further down in the tracking issue for cleaning up dead code and renaming things. Didn't want to do too much at once
